### PR TITLE
Tests cleanup

### PR DESCRIPTION
**Pull request by fbourigault**
This aim to clean the test suite to reduce tests complexity.

@stof is it what you have in mind for #96?

- [x] Rewrite MailGun tests.
- [x] Rewrite Mandrill tests.
- [x] Rewrite Postmark tests.
- [x] Rewrite SendGrid tests.

### CODE REVIEW

File: tests/Stampie/Tests/Mailer/MailGunTest.php [hunk 1]
```diff
@@ -2,120 +2,100 @@
...
 
+    /**
+     * @var MailGun
+     */
+    private $mailer;
+
+    /**
+     * @var HttpClient|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $httpClient;
```
Comment (stof) [inline tests/Stampie/Tests/Mailer/MailGunTest.php]: why not using the `$adapter` property you added in the parent ?

Comment (fbourigault) [reply tests/Stampie/Tests/Mailer/MailGunTest.php]: The idea is to remove the adapter property from the parent and keep only helper methods in the parent class. Having an HttpClient is an implementation detail.

**Comment by fbourigault** [comment]
In https://github.com/Stampie/Stampie/pull/97/commits/cf896b2c69f213f10103e132daf82f95d4a295bd I removed the `BaseMailerTest` and moved all mock related helper methods to a new `TestCase` class which can be used by mailer and non-mailer tests.

**Comment by fbourigault** [comment]
Mailer test rewrite is now done. If this fine, I will be able to go forward and kill the Mailer abstract class.

**Comment by fbourigault** [comment]
@Stampie/collaborators does it looks good to you?

**Comment by Nyholm** [comment]
Awesome

**Pull request merged**